dali/kernels/imgproc/jpeg/jpeg_distortion_gpu_test.cu

```diff
@@ -257,64 +292,76 @@ class JpegDistortionTestGPU : public ::testing::Test {
         subsample_f(tmp_cr);
       }
 
-#if DEBUG_LOGS
-      std::cout << "\nYCbCr subsampled ref:\n";
-      for (int64_t i = 0; i < npixels; i++) {
-        if (i > 0 && i % sh[1] == 0) std::cout << "\n";
-        std::cout << " " << static_cast<int>(tmp_y[i])
-                  << "-" << static_cast<int>(tmp_cb[i])
-                  << "-" << static_cast<int>(tmp_cr[i]);
-      }
-      std::cout << "\n";
-#endif
-
+      auto* sample_data = out_ref_view[sample].data;
       for (int64_t i = 0; i < npixels; i++) {
         float y = static_cast<float>(tmp_y[i]);
         float cb = static_cast<float>(tmp_cb[i]) - 128.0f;
         float cr = static_cast<float>(tmp_cr[i]) - 128.0f;
         auto r = ConvertSat<T>(y + 1.402f * cr);
         auto g = ConvertSat<T>(y - 0.34413629f * cb - 0.71413629f * cr);
         auto b = ConvertSat<T>(y + 1.772f * cb);
-        ref_output_.push_back(r);
-        ref_output_.push_back(g);
-        ref_output_.push_back(b);
+        sample_data[3*i]   = r;
+        sample_data[3*i+1] = g;
+        sample_data[3*i+2] = b;
       }
     }
   }
 
   void CalcOut_JpegCompressionDistortion() {
-    // TODO(janton): Implement JPEG encode/decode to produce the reference
-    CalcOut_ChromaSubsampleDistortion();
+    out_ref_.reshape(in_shapes_);
+    auto out_ref_view = out_ref_.cpu();
+    auto in_view_cpu = in_.cpu();
+    for (size_t i = 0; i < in_shapes_.size(); i++) {
+      auto sh = in_shapes_[i];
+      cv::Mat in_mat(sh[0], sh[1], CV_8UC3, (void *) in_view_cpu[i].data);
+
+      std::vector<uint8_t> encoded;
+
+      cv::cvtColor(in_mat, in_mat, cv::COLOR_RGB2BGR);
+      cv::imencode(".jpg", in_mat, encoded, {cv::IMWRITE_JPEG_QUALITY, ConvertSat<int>(quality_factor)});
+      cv::cvtColor(in_mat, in_mat, cv::COLOR_BGR2RGB);
```

Comment:
Do we need this 2nd conversion?

Reply:
just want to leave the input matrix as it was before (RGB).

Reply:
It is not used anymore so I think it is just redundant.